src/UserContext.js

```diff
@@ -0,0 +1,65 @@
+import { createContext, useState, useContext } from "react";
+import { dataUsers } from "./UserData";
+
+
+const UserContext = createContext();
+export const useUser = () => {
+  return useContext(UserContext);
+};
+
+export const UserProvider = ({ children }) => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [userDetails, setUserDetails] = useState(null);
+
+  const login = (username, password) => {
+    const someUser = dataUsers.find(
+      (user) => user.username === username && user.password === password
+    );
+
+    if (someUser) {
+      setIsLoggedIn(true);
+      setUserDetails(someUser);
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  const logout = () => {
+    setIsLoggedIn(false);
+    setUserDetails(null);
+  };
+
+  const addFavoriteRecipe = (recipe) => {
+    setUserDetails((prevDetails) => ({
+      ...prevDetails,
+      favoriteRecipes: prevDetails.favoriteRecipes
```

Comment:
No need to do changes here but for the future in situations like this where you want to save external data (recipes belong to recipe module), it's more common to store ids only so the data is more lightweight. Id then works as a reference, if you know what's the ids of user's favorite recipes are, you can always fetch them from the data storage where recipes are stored. 
This way, when updating recipes, you'd also only need to update them in one place. E.g. it could be that recipe with id idMeal:1234 has ingredient or image change, in the current structure, you'd need to update both recipe storage and the data here as well.